aws-api/src/main/java/com/amplifyframework/api/aws/SubscriptionOperation.java

```diff
@@ -76,14 +79,12 @@ public synchronized void start() {
             ));
             return;
         }
-        executorService.submit(() -> {
+        subscriptionFuture = executorService.submit(() -> {
             LOG.debug("Requesting subscription: " + getRequest().getContent());
             subscriptionEndpoint.requestSubscription(
+                subscriptionId,
```

Comment:
Is there a reason you need to pass in `subscriptionId`?  It seems like that could be handled internally within `SubscriptionEndpoint`.  It doesn't appear that `subscriptionId` is actually used anywhere else in `SubscriptionOperation`, except for checking if it's null in `cancel()`, which doesn't seem like a necessary check anyway.

Reply:
Nope...I reverted that change.